substrate/bin/utils/chain-spec-builder/build.rs

```diff
@@ -18,6 +18,14 @@
 
 use std::env;
 
+#[cfg(feature = "generate-readme")]
+#[allow(unused)]
+// We need this function to be compiled so that it triggers
+// the markdown compilation into the `README.md` file.
+fn compile_docs_markdown() {
```

Comment:
how/when is it called?

Reply:
When running 'cargo build/doc --features generate-readme'.

Reply:
Is  the explicit call missing in `build.rs::main()`?

Reply:
We don't need an explicit call. The docs are generated at the compile time (we don't have to call it inside the `build.rs::main()`).

Reply:
`Docify` docs excerpt:
```
While files are compiling, terminal output is produced such as:

Docifying fixtures/subfolder/file_2.md => test_bin/subfolder/file_2.md
```

Reply:
:thinking: Indeed. After some more tested I settled on this solution([e6c3ab4](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/e6c3ab4132a32faa3582daf29d7bfbe0915b69a5)) which seems to trigger the README.md update with each README.docify.md modification.

Reply:
> How this (private) function is called then? Which (cargo) components know about it?

The pre-processing logic from `compile_markdown!` macro (which is executed at compile time) will update the file. `compile_markdown` is a proc_macro (https://docs.rs/docify_macros/0.2.8/src/docify_macros/lib.rs.html#1286), so it will be expanded to rust code at compile time, that should compile (but that's irrelevant for its purpose). What's relevant is that it will execute some logic, which is exactly what updates the README.md file.